<yvals_core.h>: Could definition of _HAS_CXX23 be removed?

There's a definition of `_HAS_CXX23` in `<yvals_core.h>`:
https://github.com/microsoft/STL/blob/d6f9987d7ec694b8599fe21e87e3653c36566223/stl/inc/yvals_core.h#L382-L395

This definition was needed because `<vcruntime.h>` did not have this macro when this definition was introduced. But it seems that latest `<vcruntime.h>` does define `_HAS_CXX23`:
```
C:\Temp>type repro.cpp
#include <vcruntime.h>

#ifdef _HAS_CXX23
#error <vcruntime.h> defines _HAS_CXX23
#endif

C:\Temp>cl /std:c++latest /E .\repro.cpp > NUL
Microsoft (R) C/C++ Optimizing Compiler Version 19.30.30528 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

/std:c++latest is provided as a preview of language features from the latest C++
working draft, and we're eager to hear about bugs and suggestions for improvements.
However, note that these features are provided as-is without support, and subject
to changes or removal as the working draft evolves. See
https://go.microsoft.com/fwlink/?linkid=2045807 for details.

repro.cpp
repro.cpp(4): fatal error C1189: #error:  <vcruntime.h> defines _HAS_CXX23
```
(also https://www.godbolt.ms/z/nMHsE5)

... so the abovementioned lines in `<yvals_core.h>` are no longer used and can be deleted.

(I did not submit a pull request myself because I think this could be a "good first issue".)

--- a/stl/inc/yvals_core.h
+++ b/stl/inc/yvals_core.h
@@ -380,7 +380,6 @@
 // * unique_copy
 
 #include <vcruntime.h>
-
 #include <xkeycheck.h> // The _HAS_CXX tags must be defined before including this.
 
 #ifndef _STL_WARNING_LEVEL

--- a/stl/inc/yvals_core.h
+++ b/stl/inc/yvals_core.h
@@ -381,19 +381,6 @@
 
 #include <vcruntime.h>
 
-// TRANSITION, <vcruntime.h> should define _HAS_CXX23
-#ifndef _HAS_CXX23
-#if _HAS_CXX20 && (defined(_MSVC_LANG) && _MSVC_LANG > 202002L || __cplusplus > 202002L)
-#define _HAS_CXX23 1
-#else
-#define _HAS_CXX23 0
-#endif
-#endif // _HAS_CXX23
-
-#if _HAS_CXX23 && !_HAS_CXX20
-#error _HAS_CXX23 must imply _HAS_CXX20.
-#endif
-
 #include <xkeycheck.h> // The _HAS_CXX tags must be defined before including this.
 
 #ifndef _STL_WARNING_LEVEL

